priv/script.sh

```diff
@@ -2,441 +2,489 @@
-postgres_version=15.1
+if [ -z "$HOME" ]; then
+  set +e
+  if [ "$os_type" = Linux ]; then
+    HOME="$(getent passwd "$USER" 2>/dev/null | cut -d: -f6)"
+  elif [ "$os_type" = macOS ]; then
+    HOME="$(dscl . -read /Users/"$USER" NFSHomeDirectory)"
+  fi
+
+  # If neither of the above commands works, fall back to `~$USER`.
+  HOME="${HOME:-$(eval echo ~"$USER")}"
+  set -e
+fi
+
+# Disable colour output if NO_COLOR is set to any value at all.
+# https://no-color.org
+if [ -n "${NO_COLOR:-}" ]; then
+  bold=''
+  normal=''
+  bblue=''
+  cyan=''
+elif has tput; then
+  # Prefer using `tput` for highlighting over manual ANSI colour codes. This
+  # would allow xterm 256 colour as well.
+  bold="$(tput bold)"
+  normal="$(tput sgr0)"
+  bblue="$(tput setaf 11)"
+  cyan="$(tput setaf 6)"
+else
+  bold='\033[1m'
+  normal='\033[0m'
+  bblue='\033[1;34m'
+  cyan='\033[0;36m'
+fi
+
+phx_tools_path="$HOME"/.config/phx_tools
+phx_tools_sh_path="$phx_tools_path"/phx_tools.sh
+phx_tools_fish_path="$phx_tools_path"/phx_tools.fish
+homebrew_path=
 
 case "${SHELL:-}" in
-*bash)
-    current_shell="bash"
-    config_file="$HOME/.bashrc"
-    ;;
-*fish)
-    current_shell="fish"
-    config_file="$HOME/.config/fish/config.fish"
-    mkdir -p "$(dirname "$config_file")"
-    ;;
-*zsh)
-    current_shell="zsh"
-    config_file="$HOME/.zshrc"
-    ;;
+*/bash | */fish | */zsh) : ;;
+'')
+  printf >&2 "Cannot discover shell, \$SHELL is unset.\n"
+  exit 1
+  ;;
 *)
-    printf "Unsupported shell: %s\n" "$SHELL"
-    exit 1
-    ;;
+  printf >&2 "Unsupported shell: %s\n" "$SHELL"
+  exit 1
+  ;;
 esac
 
-# Add OS detection
-OS="$(uname -s)"
-
 get_package_manager() {
-  if command -v apt-get >/dev/null; then
-    echo "apt"
-  elif command -v dnf >/dev/null; then
-    echo "dnf"
-  elif command -v pacman >/dev/null; then
-    echo "pacman"
-  elif command -v apk >/dev/null; then
-    echo "apk"
+  if has apt-get; then
+    echo apt
+  elif has dnf; then
+    echo dnf
+  elif has pacman; then
+    echo pacman
+  elif has apk; then
+    echo apk
   else
-    printf "Unsupported package manager. This script requires apt, dnf, pacman, or apk.\n"
+    printf >&2 "Unsupported package manager.\n"
+    printf >&2 "This script requires apt, dnf, pacman, or apk.\n"
     exit 1
   fi
 }
 
-case "${OS}" in
-  Linux*)
-    os_type=Linux
-    package_manager=$(get_package_manager)
-    ;;
-  Darwin*)
-    os_type=macOS
-    ;;
+if [ "$os_type" = macOS ]; then
+  package_manager=homebrew
+
+  if [ "$(uname -m)" = arm64 ]; then
+    homebrew_path=/opt/homebrew/bin
+  else
+    homebrew_path=/usr/local/bin
+  fi
+else
+  package_manager="$(get_package_manager)"
+fi
+
+already_installed() {
+  case "$1" in
+  Elixir) mise_has_version elixir "$elixir_version" ;;
+  Erlang) mise_has_version erl "$erlang_version" ;;
+  git) has git ;;
+  Homebrew) has brew ;;
+  mise) has mise ;;
+  Phoenix) has mix && mix phx.new --version >/dev/null 2>&1 ;;
+  PostgreSQL) mise_has initdb ;;
```

Comment:
This should *probably* be `has initdb || mise_has initdb`. I personally install PostgreSQL through MacPorts, which gives me many of the benefits of a mise installation in terms of version stability.